src/test/java/net/masterthought/cucumber/ConfigurationTest.java

```diff
@@ -186,6 +186,90 @@ public void getDirectorySuffixWithSeparatorForEmptySuffix_ReturnsEmptyString() {
         assertThat(configuration.getDirectorySuffixWithSeparator()).isEqualTo("");
     }
 
+    @Test
+    public void getQualifier_ReturnsQualifierWhenSet() {
+
+        // given
+        String jsonFile = "test";
+        String qualifier = "test";
+        Configuration configuration = new Configuration(outputDirectory, projectName);
+
+        // when
+        configuration.setQualifier(jsonFile, qualifier);
+
+        // then
+        assertThat(configuration.getQualifier(jsonFile)).isEqualTo(qualifier);
+    }
+
+    @Test
+    public void getQualifier_ReturnsNullWhenNotSet() {
+
+        // given
+        String jsonFile = "test";
+        Configuration configuration = new Configuration(outputDirectory, projectName);
+
+        // then
+        assertThat(configuration.getQualifier(jsonFile)).isNull();
+    }
+
+    @Test
+    public void getQualifier_ReturnsNullWhenSetThenRemoved() {
+
+        // given
+        String jsonFile = "test";
+        String qualifier = "test";
+        Configuration configuration = new Configuration(outputDirectory, projectName);
+
+        // when
+        configuration.setQualifier(jsonFile, qualifier);
+        configuration.removeQualifier(jsonFile);
+
+        // then
+        assertThat(configuration.getQualifier(jsonFile)).isNull();
+    }
+
+    @Test
+    public void isQualifierSet_ReturnsTrueWhenSet() {
+
+        // given
+        String jsonFile = "test";
+        String qualifier = "test";
+        Configuration configuration = new Configuration(outputDirectory, projectName);
+
+        // when
+        configuration.setQualifier(jsonFile, qualifier);
+
+        // then
+        assertThat(configuration.isQualifierSet(jsonFile)).isTrue();
+    }
+
+    @Test
+    public void isQualifierSet_ReturnsTrueWhenNotSet() {
+
+        // given
+        String jsonFile = "test";
+        Configuration configuration = new Configuration(outputDirectory, projectName);
+
+        // then
+        assertThat(configuration.isQualifierSet(jsonFile)).isFalse();
+    }
+
+    @Test
+    public void isQualifierSet_ReturnsTrueWhenSetThenRemoved() {
+
+        // given
+        String jsonFile = "test";
+        String qualifier = "test";
+        Configuration configuration = new Configuration(outputDirectory, projectName);
+
+        // when
+        configuration.setQualifier(jsonFile, qualifier);
+        configuration.removeQualifier(jsonFile);
+
+        // then
+        assertThat(configuration.isQualifierSet(jsonFile)).isFalse();
+    }
+
```

Comment:
I like that above - great to have it.

I need one more to test with https://github.com/damianszczepanik/cucumber-reporting/blob/master/src/test/java/net/masterthought/cucumber/generators/integrations/FeaturesOverviewPageIntegrationTest.java#L149 that the effects is present in overview page as well. Copy mentioned test, set qualifier and examine results

Reply:
Added an integration test `FeaturesOverviewPageIntegrationTest.generatePage_WithSpecifiedQualifier_generatesStatsTableBody` checking that the qualifier has been set and used correctly.